### PR TITLE
Fixes #10 Fix the test issues of virtualization sdk on windows

### LINKED ISSUE #10: Fix the test issues of virtualization sdk on windows
virtualization sdk build is failing on windows as existing test cases fail. Need to fix them before enabling the windows-latest platform in the pre-commit Github Action.

┆Issue is synchronized with this [Jira Story](https://jira.delphix.com/browse/PYT-1088) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by muralinimmagadda**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [X ] Tests for the changes have been added (for bug fixes / features)
- [ X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [X ] Build was run locally and any changes were pushed
- [X ] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [X ] Other (please describe): 

## What is the current behavior?

unit tests currently failing on windows
Issue Number: 10

## What is the new behavior?

Github actions for virtualization-sdk now runs on windows too.

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [X ] No

### CODE REVIEW

File: tools/src/test/python/dlpx/virtualization/_internal/test_file_util.py [hunk 1]
```diff
@@ -80,13 +81,13 @@ def test_get_src_dir_path_is_dir_fail(mock_existing_path,
     @mock.patch('os.path.isabs', return_value=False)
     @pytest.mark.parametrize(
         'plugin_config_file_path, src_dir_path',
-        [(os.path.join(os.getenv('HOME'), 'plugin/file_name'), '.'),
+        [('plugin/file_name', '.'),
          ('/mongo/file_name', '/src'), ('/plugin/mongo/file_name', '/plugin'),
          ('/plugin/file_name', '/plugin/src/../..')])
```
Comment (fdrozdowski) [inline tools/src/test/python/dlpx/virtualization/_internal/test_file_util.py:86]: Is there a reason we don't use `os.path.join` for these paths? Is it because we use `os.path.join` in line 90 instead?

File: tools/src/main/python/dlpx/virtualization/_internal/plugin_importer.py [hunk 2]
```diff
@@ -184,6 +170,20 @@ def __check_for_required_methods(self):
...
+    warnings = defaultdict(list)
+    while not queue.empty():
+        q_item = queue.get()
+        if 'manifest' in q_item:
+            manifest = q_item['manifest']
+        else:
+            key = list(q_item.keys())[0]
+            warnings[key].append(q_item[key])
+
+    return manifest, warnings
```
Comment (fdrozdowski) [inline tools/src/main/python/dlpx/virtualization/_internal/plugin_importer.py:184]: What's the reason we made this method non-static?

File: tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py [hunk 3]
```diff
@@ -499,7 +500,26 @@ def test_schema_file_bad_permission(mock_generate_python,
...
+            # admins, domain, type = win32security.LookupAccountName("", "Administrators")
+            # user, domain, type = win32security.LookupAccountName("", win32api.GetUserName())
+
+            # dacl = win32security.ACL()
+            # dacl.AddAccessAllowedAce(win32security.ACL_REVISION, con.FAILED_ACCESS_ACE_FLAG, everyone)
+            # dacl.AddAccessAllowedAce(win32security.ACL_REVISION, con.FAILED_ACCESS_ACE_FLAG, user)
+            # dacl.AddAccessAllowedAce(win32security.ACL_REVISION, con.FAILED_ACCESS_ACE_FLAG, admins)
+
+            # sd.SetSecurityDescriptorDacl(1, dacl, 0)
+            # win32security.SetFileSecurity(schema_file, win32security.DACL_SECURITY_INFORMATION, sd)
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/commands/test_build.py:520]: did you mean to leave these as comments? maybe for future reference? seems like if we're just gonna skip we might want to remove these comments?

File: tools/src/test/python/dlpx/virtualization/_internal/commands/test_codegen.py [hunk 4]
```diff
@@ -363,12 +363,15 @@ def test_copy_generated_to_dir_success(tmpdir, popen_helper):
...
         dst_dir = tmpdir.strpath
 
         with pytest.raises(OSError) as err_info:
             codegen._copy_generated_to_dir(src_dir, dst_dir)
 
-        assert err_info.value.strerror == 'No such file or directory'
-        assert err_info.value.filename.startswith('/not/a/real/dir')
+        assert (
+            err_info.value.strerror == 'No such file or directory' or
+            err_info.value.strerror == 'The system cannot find the path specified'
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/commands/test_codegen.py:375]: is one of these the windows error and the other the unix? can we just leave a comment saying which os throws what so that in the future we know why we'd expect different error messages?

Or better yet maybe we can assert if os.name == 'nt' (? im just using the if statement you had above) and the expected error message that would make it explicit so that if something fails we'll know pretty quickly why?

File: tools/src/test/python/dlpx/virtualization/_internal/test_file_util.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 #
 
 import os
+import ntpath
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/test_file_util.py:6]: maybe im missing it but it doesn't look like we're using ntpath as an import anymore?

File: tools/src/test/python/dlpx/virtualization/_internal/test_plugin_importer.py [hunk 6]
```diff
@@ -47,45 +49,42 @@ def test_get_plugin_manifest(mock_import, src_dir, plugin_type,
...
+        plugin_importer._get_manifest(queue, src_dir, entry_point_module, entry_point_object, plugin_type, False)
+        manifest, warnings = plugin_importer._parse_queue(queue)
 
         #
         # If module_content is None, importer does not perform any validations
         # and just does a return. So result should have an empty manifest and
         # assert to make sure it is the case.
         #
-        assert result.plugin_manifest == {}
+        assert manifest == {}
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/test_plugin_importer.py:73]: could you leave comments explaining why this change was needed? it's odd to me that we'd have to do this so maybe I'm missing something

File: tools/src/test/python/dlpx/virtualization/_internal/test_schema_validator.py [hunk 7]
```diff
@@ -21,7 +21,7 @@ def test_bad_meta_schema(schema_file, tmpdir, schema_filename):
             validator.validate()
 
         message = err_info.value.message
-        assert ('Failed to load schemas because {!r} is not a valid json file.'
+        assert ('Failed to load schemas because \'{}\' is not a valid json file.'
                 ' Error: Extra data: line 2 column 1 - line 2 column 9'
                 ' (char 19 - 27)'.format(schema_file)) in message
```
Comment (nhlien93) [inline tools/src/test/python/dlpx/virtualization/_internal/test_schema_validator.py:26]: per our style guidelines, if your string will have single quotes you'll want to switch this to use double quotes rather than escape the single quotes, so:
```
assert ("Failed to load schemas because '{}' is not a valid json file."
             " Error: Extra data: line 2 column 1 - line 2 column 9"
             " (char 19 - 27)".format(schema_file)) in message
```

**Comment by muralinimmagadda** [comment]
Messed up my fork trying to raise another parallel pullrequest. sorry for the incovenience. will close this pull request and raise new ones.
